app/schema_validation/__init__.py

```diff
@@ -60,10 +63,17 @@ def validate(json_to_validate, schema):
     validator = Draft7Validator(schema, format_checker=format_checker)
     errors = list(validator.iter_errors(json_to_validate))
     if len(errors) > 0:
+        if isinstance(json_to_validate, dict) and "personalisation" in json_to_validate:
+            if isinstance(json_to_validate["personalisation"], str):
+                json_to_validate["personalisation"] = "<redacted>"
+            elif isinstance(json_to_validate["personalisation"], dict):
+                for key in json_to_validate["personalisation"]:
+                    json_to_validate["personalisation"][key] = "<redacted>"
```

Comment:
Dictionary comprehension please. This isn't pythonic.

Something like this please:
```
json_to_validate['personalisation'] = {k: '<redacted>' for k, v in json_to_validate['personalisation'].items()}
```

Reply:
changed